libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -304,6 +364,32 @@ void HTTPClient::setTimeout(uint16_t timeout)
     }
 }
 
+bool HTTPClient::setRootCA(uint8_t * cert, size_t size)
+{
+    bool result = false;
+    if(_tcp && (_protocol == "https" || _port == 443)) {
+        auto client = static_cast<WiFiClientSecure*>(_tcp.get());
```

Comment:
The approach you suggest would be more expensive as it requires storing the cert, which can be quite large.